### DISCUSSION TITLE
query associative aray with multiple keys with a single key

### DISCUSSION THREAD

**Discussion by vimalk78** [bpftrace/bpftrace#2468] [Q&A] [answered]
```
@my_map[key1,key2] = val1;
```

while querying, i only have first key. can i query it with only one key, ignoring other key?
```
if ( @my_map[key1, _ ] == val2 )
{ 
}
```

**Comment by fbs** [comment, marked as answer]
You cannot. Multiple keys is a not a BPF concept, its bpftrace. 

The core function is this, it just takes an arbitrary length key (length is set during map creation)
```
                  int
                  bpf_lookup_elem(int fd, const void *key, void *value)
                  {
                      union bpf_attr attr = {
                          .map_fd = fd,
                          .key    = ptr_to_u64(key),
                          .value  = ptr_to_u64(value),
                      };

                      return bpf(BPF_MAP_LOOKUP_ELEM, &attr, sizeof(attr));
                  }
```

So for this to work what you really want is to iterate over every key in the map and then test whether the key matches. I think this would be possible to do in BPF, just not in bpftrace (yet). It also scales with map entries so it can get slow fast.

A better approach would be a different data structure but thats hard to do too.

tl;dr: no

**Reply by vimalk78** [reply]
thanks @fbs

**Reply by viktormalik** [reply]
In fact, once #2321 is merged, it'll be possible to delete map entries based on the value of a single part of a multi-value key.
